### ISSUE TITLE
Not all export filters seem to be able to handle private:stream

### ISSUE THREAD

**Issue by jerbob92** [unoconv/unoserver#117]
Hi there!

When executing the following command:

```
unoconvert --output-filter "DocBook File" --convert-to xml - - < infile.rtf
```

To convert an RTF file to a DocBook XML File, and sending the output to stdout (really specific, I know), the following error occurs: 

```
uno.com.sun.star.uno.RuntimeException: illegal object given! at /home/buildozer/aports/community/libreoffice/src/libreoffice-7.6.3.1/stoc/source/corereflection/crefl.cxx:320
```

And the LibreOffice process seems to crash.

When outputting to a file path:

```
unoconvert --output-filter "DocBook File" --convert-to xml - outfile.xml < infile.rtf
```

Everything seems to work fine! 

When looking at the code the only difference seems to be the usage of `OutputStream` in the first command.

Any idea what might be going on here? Thanks in advance!

**Comment by regebro** [comment]
Yeah, you are right, I get the same error. That's amazingly annoying. The only way I can see to fix that is to use temporary files in the server part, which is doable, but really shouldn't be needed.

**Comment by jerbob92** [comment]
That's how we have solved it now indeed. 
Would the next step be reporting an issue with LibreOffice? Or does this issue originate in the uno/unoserver package?

**Comment by regebro** [comment]
It's Libreoffice, and getting bugs fixed there takes time, they definitely need more people helping actually fix bugs, which I unfortunately don't have the time or language experience to do.

**Comment by jerbob92** [comment]
Same for me, I don't see myself diving into the LibreOffice codebase right now.

I have reported the issue here: https://bugs.documentfoundation.org/show_bug.cgi?id=161416

**Comment by whitehat101** [comment]
Given this comment [161416#c3](https://bugs.documentfoundation.org/show_bug.cgi?id=161416#c3)
>Noel Grandin 2024-06-27 09:54:35 UTC
>Reasonable feature request, not sure when/if anyone will around to it, however.

Does it make sense to add an option to this project to use a tempfile instead of a stream?

We could also add a manual blocklist of known filters with issues.

Or first try the stream export and if we get back an empty byte string retry with a tempfile.

We could combine the two, and keep an automatic blocklist in memory, so we only try the stream once per filter per unoserver process.

Since upstream doesn't seem to have the bandwidth/motivation to work on this bug, it would be nice to band-aid it in this app. A fix in uno-server is likely easier than patching an unknown number of filters upsteam, too.

@regebro Do you have any thoughts or preferences on long-term workaround?
